httpclient5/src/test/java/org/apache/hc/client5/http/impl/auth/TestAuthChallengeParser.java

```diff
@@ -321,12 +323,26 @@ public void testParseNTLMAuthChallenge() throws Exception {
         Assert.assertEquals(null, challenge1.getValue());
     }
 
-    private static void assertNameValuePair (
-            final NameValuePair expected,
-            final NameValuePair result) {
-        Assert.assertNotNull(result);
-        Assert.assertEquals(expected.getName(), result.getName());
-        Assert.assertEquals(expected.getValue(), result.getValue());
+    @Test
+    public void testParseParameterAndToken68AuthChallengeMix() throws Exception {
+        final CharArrayBuffer buffer = new CharArrayBuffer(64);
+        buffer.append("scheme1 aaaa  , scheme2 aaaa==,  scheme3 aaaa=aaaa, scheme4 aaaa=");
+        final ParserCursor cursor = new ParserCursor(0, buffer.length());
+        final List<AuthChallenge> challenges = parser.parse(ChallengeType.TARGET, buffer, cursor);
+        Assert.assertNotNull(challenges);
+        Assert.assertEquals(4, challenges.size());
+        final AuthChallenge challenge1 = challenges.get(0);
+        Assert.assertThat(challenge1.getSchemeName(), CoreMatchers.equalTo("scheme1"));
+        Assert.assertThat(challenge1.getValue(), CoreMatchers.equalTo("aaaa"));
+        final AuthChallenge challenge2 = challenges.get(1);
+        Assert.assertThat(challenge2.getSchemeName(), CoreMatchers.equalTo("scheme2"));
+        Assert.assertThat(challenge2.getValue(), CoreMatchers.equalTo("aaaa=="));
+        final AuthChallenge challenge3 = challenges.get(2);
+        Assert.assertThat(challenge3.getSchemeName(), CoreMatchers.equalTo("scheme3"));
+        Assert.assertThat(challenge3.getValue(), CoreMatchers.nullValue());
```

Comment:
I don't understand this. Param is there no param map with the key `aaaa`?  I this supported to match token68 only?

Reply:
@michael-o This is precisely the problem as there is no way of telling if you are looking at a parameter or at a padded token68. The current solution is to assume that anything looking ambiguous is likely a token86 but `aaaa=aaaa` definitely more likely a parameter.

Reply:
How can this logically be padding, because the token is supported to terminal with an equals sign. If it does not, it param=value. Isn't it?

Reply:
@michael-o But this is what I am saying. It is a parameter (name/value pair).

Reply:
Yes, I know, but I don't see this in the test. That's the issue I have. Am I blind here?

Reply:
@michael-o This is a header value used by the test: `scheme1 aaaa  , scheme2 aaaa==,  scheme3 aaaa=aaaa, scheme4 aaaa=`. Schemes 1, 2 and 4 are token68 based and scheme 3 is parameter based. What is still missing?

Reply:
Apololiges, I misread the test. It looks fine, but aI would add testver for the params that `aaaa=aaaa` has been parsed as such from the input mess.